### PR TITLE
Fix decoding issue with Arrays

### PR DESCRIPTION
**Pull request by doomspork**

### CODE REVIEW

File: lib/avro_ex/decode.ex [hunk 1]
```diff
@@ -6,8 +6,7 @@ defmodule AvroEx.Decode do
 
   @type reason :: term
 
-  def decode(%Schema{schema: schema, context: context}, avro_message)
-      when is_binary(avro_message) do
+  def decode(%Schema{schema: schema, context: context}, avro_message) when is_binary(avro_message) do
```
Comment (doomspork) [inline lib/avro_ex/decode.ex]: I will put this back 👍

File: lib/avro_ex/decode.ex [hunk 2]
```diff
@@ -169,8 +168,10 @@ defmodule AvroEx.Decode do
   def do_decode(%Array{items: item_schema}, %Context{} = context, data) when is_binary(data) do
     {count, buffer} = do_decode(%Primitive{type: :long}, context, data)
 
+    times = if count > 0, do: 1..count, else: []
```
Comment (doomspork) [inline lib/avro_ex/decode.ex:172]: I do not love this at all. I would like something better. Ideas?

File: test/decode_test.exs [hunk 3]
```diff
@@ -93,7 +93,7 @@ defmodule AvroEx.Decode.Test do
       {:ok, encoded_message} = AvroEx.encode(schema, %{"a" => 1, "b" => 2, "e" => "Hello world!"})
 
       assert {:ok, %{"a" => 1, "b" => 2, "e" => "Hello world!"}} =
-               @test_module.decode(schema, encoded_message)
+        @test_module.decode(schema, encoded_message)
```
Comment (doomspork) [inline test/decode_test.exs]: I'll re-tab this.

File: test/decode_test.exs [hunk 4]
```diff
@@ -161,6 +161,43 @@ defmodule AvroEx.Decode.Test do
       {:ok, encoded_sha} = AvroEx.encode(schema, sha)
       assert {:ok, ^sha} = @test_module.decode(schema, encoded_sha)
     end
+
+    test "record with empty array of records" do
+      {:ok, schema} = AvroEx.parse_schema(~S(
+        {
```
Comment (doomspork) [inline test/decode_test.exs:167]: I expanded this out for now so we could all see what the schema was.

File: lib/avro_ex/decode.ex (hunk 2 above)
Comment (michaelstalker) [reply lib/avro_ex/decode.ex:172]: I can't think of a better solution. Extracting the reduction into a helper function is not very clean, either. I'm okay with this. The `times` name doesn't quite fit the data structures, but it fits the usage.

**Review by michaelstalker — APPROVED** [review]

**Pull request merged**